javascript/packages/linter/test/rules/html-no-empty-attributes.test.ts

```diff
@@ -0,0 +1,291 @@
+
+    const linter = new Linter(Herb, [HTMLNoEmptyAttributesRule])
+    const lintResult = linter.lint(html)
+
+    expect(lintResult.errors).toBe(0)
+    expect(lintResult.warnings).toBe(1)
+
+    expect(lintResult.offenses[0].message).toBe('Attribute `class` must not be empty. Either provide a meaningful value or remove the attribute entirely.')
+    expect(lintResult.offenses[0].severity).toBe("warning")
+  })
+
+  test("fails for empty name attribute", () => {
+    const html = '<input name="">'
+
+    const linter = new Linter(Herb, [HTMLNoEmptyAttributesRule])
+    const lintResult = linter.lint(html)
+
+    expect(lintResult.errors).toBe(0)
+    expect(lintResult.warnings).toBe(1)
+
+    expect(lintResult.offenses[0].message).toBe('Attribute `name` must not be empty. Either provide a meaningful value or remove the attribute entirely.')
+    expect(lintResult.offenses[0].severity).toBe("warning")
+  })
+
+  test("fails for empty for attribute", () => {
+    const html = '<label for="">Label</label>'
+
+    const linter = new Linter(Herb, [HTMLNoEmptyAttributesRule])
+    const lintResult = linter.lint(html)
+
+    expect(lintResult.errors).toBe(0)
+    expect(lintResult.warnings).toBe(1)
+
+    expect(lintResult.offenses[0].message).toBe('Attribute `for` must not be empty. Either provide a meaningful value or remove the attribute entirely.')
+    expect(lintResult.offenses[0].severity).toBe("warning")
+  })
+
+  test("fails for empty src attribute", () => {
+    const html = '<img src="" alt="Image">'
+
+    const linter = new Linter(Herb, [HTMLNoEmptyAttributesRule])
+    const lintResult = linter.lint(html)
+
+    expect(lintResult.errors).toBe(0)
+    expect(lintResult.warnings).toBe(1)
+
+    expect(lintResult.offenses[0].message).toBe('Attribute `src` must not be empty. Either provide a meaningful value or remove the attribute entirely.')
+    expect(lintResult.offenses[0].severity).toBe("warning")
+  })
+
+  test("fails for empty href attribute", () => {
+    const html = '<a href="">Link</a>'
+
+    const linter = new Linter(Herb, [HTMLNoEmptyAttributesRule])
+    const lintResult = linter.lint(html)
+
+    expect(lintResult.errors).toBe(0)
+    expect(lintResult.warnings).toBe(1)
+
+    expect(lintResult.offenses[0].message).toBe('Attribute `href` must not be empty. Either provide a meaningful value or remove the attribute entirely.')
+    expect(lintResult.offenses[0].severity).toBe("warning")
+  })
+
+  test("fails for empty alt attribute", () => {
+    const html = '<img src="image.png" alt="">'
+
+    const linter = new Linter(Herb, [HTMLNoEmptyAttributesRule])
+    const lintResult = linter.lint(html)
+
+    expect(lintResult.errors).toBe(0)
+    expect(lintResult.warnings).toBe(1)
+
+    expect(lintResult.offenses[0].message).toBe('Attribute `alt` must not be empty. Either provide a meaningful value or remove the attribute entirely.')
+    expect(lintResult.offenses[0].severity).toBe("warning")
+  })
```

Comment:
I believe this one conflicts with the [`html-img-require-alt`](https://herb-tools.dev/linter/rules/html-img-require-alt) rule. I guess we have to allow empty `alt=""` for that reason.